pkg/ingester/ingester.go

```diff
@@ -3487,14 +3491,26 @@ func (i *Ingester) checkAvailable() error {
 	return newUnavailableError(s)
 }
 
+// PushToStorage is a low-level implementation of Push() for block storage.
+func (i *Ingester) PushToStorage(ctx context.Context, req *mimirpb.WriteRequest) error {
+	err := i.PushWithCleanup(ctx, req, func() { mimirpb.ReuseSlice(req.Timeseries) })
+	if err != nil {
+		return i.mapPushErrorToErrorWithStatus(err)
+	}
+	return nil
+}
+
 // Push implements client.IngesterServer
```

Comment:
We could update this comment to mention that this gRPC method. Maybe something like

```suggestion
// Push implements client.IngesterServer, which is registered into gRPC server. 
```